File: CPPSolutions/LeetCode/String/M3174_清除数字.cpp
#include <bits/stdc++.h>

using namespace std;

/**
 * @author Hyperspace
 * @date 2024/09/05
 * @file M3174_清除数字.cpp
 */
class Solution {
public:
    string clearDigits(string s) {
        string ans;
        for (auto c: s) {
            if (!ans.empty() && isdigit(c)) {
                ans.pop_back();
                continue;
            }
            ans += c;
        }

        return ans;
    }
};

File: GolangSolutions/String/M3174_清除数字.go
/*
	@author Hyperspace
	@date 2024/09/05
	@file M3174_清除数字.go
*/

package String

import "unicode"

func clearDigits(s string) string {
	ans := []rune{}
	for _, c := range s {
		if len(ans) > 0 && unicode.IsDigit(c) {
			ans = ans[:len(ans)-1]
			continue
		}
		ans = append(ans, c)
	}

	return string(ans)
}
